docs/development.md

```diff
@@ -34,49 +34,109 @@ To summarize the steps below, everything you need to do:
-extension tool. The configuration for the tool can be found under
-`workspace-configuration/extensions`. Make sure that you use at least version
-4.0.1 of the extension tool.
-
-Here you need to add the following configuration for the `extensions-maven-plugin`
-```xml
-<configuration>
-  <projectRoot>../..</projectRoot>
-  <extensionsRoot>modules/extensions</extensionsRoot>
-  <extensionPointsPath>modules/extension-config</extensionPointsPath>
-</configuration>
+
+Thus, `1907.1-1` signals compatibility with CMCC 1907.1 and is the first
+version of this GlobalLink workspace. `1907.1-2` is a patch version for
+version `1907.1-1`, which is based on the same CMCC version, but for example
+contains bug fixes.
+
+## Adding GCC Adapter to the Blueprint
+
+There are many approaches for integrating the extension into
+the Blueprint. Each one comes with its own pros and cons depending on your use
+case.
+
+With `git subtree` you can easily change the adapter code and cherry-pick important
+fixes of upcoming releases without having to create a fork of the adapter itself.
+This is one of the main advantages of the subtree approach over `git submodule`. 
+Contributing back upstream is however slightly more complicated.
+
+Feel free to choose the strategy that fits your needs best. For example:
+
+* As a Git Subtree from the workspace root (recommended)
+   ```bash
+   $ mkdir -p modules/extensions
+   $ # Add sub-project as a remote to enable short form
+   $ git remote add -f gcc https://github.com/CoreMedia/coremedia-globallink-connect-integration.git
+   $ git subtree add --prefix modules/extensions/gcc gcc main --squash
+   $ # For example - update the sub-project at a later date...
+   $ git fetch gcc main
+   $ git subtree pull --prefix modules/extensions/gcc gcc main --squash
+   ```
+  See [Issue 28](https://github.com/CoreMedia/coremedia-globallink-connect-integration/issues/28) for a more detailed description of a similar approach.
+* Or as a Git Submodule from the workspace root
+   ```bash
+   $ git submodule add https://github.com/CoreMedia/coremedia-globallink-connect-integration.git modules/extensions/gcc
+   $ git submodule update --init --recursive
+   $ cd modules/extensions/gcc
+   $ git checkout <release-tag>
+   $ cd ..
+   $ # Add and commit .gitmodules and current HEAD of submodule 
+   $ git add .
+   $ git commit -m "Initial integration of submodule based on <release-tag>"
+   ```
+  
+If you want to contribute to this project - which we hope for - you 
+need to fork the project. For example, with the `git subtree` approach, pushing 
+to your fork could look as follows:
+```bash
+$ # Add your fork as another remote
+$ git remote add -f my-gcc https://github.com/my-company/coremedia-globallink-connect-extended.git
+$ # For example - update the sub-project at a later date...
+$ git subtree push --prefix=modules/extensions/gcc my-gcc main
 ```
+You can then send us the corresponding pull request.
```

Comment:
Text-Only Readability:

````suggestion
to your fork could look as follows:

```bash
$ # Add your fork as another remote
$ git remote add -f my-gcc https://github.com/my-company/coremedia-globallink-connect-extended.git
$ # For example - update the sub-project at a later date...
$ git subtree push --prefix=modules/extensions/gcc my-gcc main
```

You can then send us the corresponding pull request.
````